### DISCUSSION TITLE
How to  query foreign tables across different schemas using javascript sdk ?

### DISCUSSION THREAD

**Discussion by jiwansvmx** [supabase/supabase#14529] [Questions] [answered]
I apologize upfront if this has been asked before - I searched through the issues and couldn't find a matching issue.

Here is what I am trying to do:

I have the following table 
```sql
create table
  public.content_version (
    id uuid not null default gen_random_uuid (),
    created_at timestamp with time zone null default now(),
    updated_at timestamp with time zone null,
    name text null,
    description text null,
    storage_id uuid null,
    thumb_id uuid null,
    constraint ContentVersion_pkey primary key (id),
    constraint content_version_storage_id_fkey foreign key (storage_id) references storage.objects (id) on delete cascade,
    constraint content_version_thumb_id_fkey foreign key (thumb_id) references storage.objects (id) on delete cascade
  ) tablespace pg_default;
```

I initialized the js sdk as follows

```ts
const supabase = createClient(PROJECT_URL, ANON_PUBLIC_KEY);
```

Following the examples in the [api docs](https://supabase.com/docs/reference/javascript/select), I am trying to select fields from content_version and include some fields from the foreign (`stroage.objects`)  with the following code

```ts
const {data: contentVersions, error} = await apiClient
          .from('content_version')
          .select(
            'name, storage_id, storage.objects(bucket_id, name, path_tokens)',
          );
```

which gives the following error 

```json
{"code": "PGRST100", "details": "unexpected '.' expecting letter, digit, \"-\", \"!\", \"(\", \"->>\", \"->\", \"::\", \")\", \",\" or end of input", "hint": null, "message": "\"failed to parse select parameter (name,storage_id,storage.objects(bucket_id,name,path_tokens))\" (line 1, column 24)"}
```

I am not sure what I am missing. Perhaps querying foreign tables across different schemas is not supported ? 

Essentially I want to associate a `content_version` row with a `file` ( row `storage.objects`) and when I query `content_version` i want to retrieve the path of associated file.

How do I go about this ?

PS: I have the necessary RLS polices configured as well as exposed both public, storage schema.

Thank you very much!!

**Comment by GaryAustin1** [comment, marked as answer]
You can't work with multiple schemas at once with the API.  The schema is stored in a header and can be changed when the client is setup.  It defaults to public.

You would need to handle this in sql and call an rpc function or use a view.

**Reply by jiwansvmx** [reply]
Go it! Thanks for answering.
